homeassistant/components/automate/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow for Automate Pulse Hub v2 integration."""
+import logging
+
+import aiopulse2
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required("host"): str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect to the hub.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    hub = aiopulse2.Hub(data["host"])
+    try:
+        await hub.test()
+    except Exception as err:
+        raise CannotConnect(str(err))  # pylint: disable=raise-missing-from
```

Comment:
Why repack this exception?

Reply:
It appeared to be a common pattern looking at other code (that's how I figured much of this out), I'll lightly refactor for further feedback, but will still want the general try catch to display feedback to the user (from the previous comment).